Microsoft.NET.Build.Containers/Image.cs

```diff
@@ -220,6 +220,10 @@ public void Label(string name, string value)
 
     public void AddEnvironmentVariable(string envVarName, string value)
     {
+        if (environmentVariables.ContainsKey(envVarName))
```

Comment:
We should overwrite here and make folks aware that this is the semantic. Otherwise, people can't do things like overwriting the `ASPNETCORE_URLS` variable on their own, since the `dotnet/runtime-deps` container includes that definition by default.